packages/client/src/MetricsPublisher.ts

```diff
@@ -0,0 +1,96 @@
+import { scoped, Lifecycle, inject } from 'tsyringe'
+import { StreamrClientEventEmitter } from './events'
+import { DestroySignal } from './DestroySignal'
+import { MetricsReport } from 'streamr-network'
+import { BrubeckNode, getEthereumAddressFromNodeId } from './BrubeckNode'
+import { Publisher } from './publish/Publisher'
+import { ConfigInjectionToken, MetricsPeriodConfig, StrictStreamrClientConfig } from './Config'
+
+const DEFAULT_PERIODS = [ 
+    {
+        "duration": 5000,
+        "streamId": "streamr.eth/metrics/nodes/firehose/sec"
+    },
+    {
+        "duration": 60000,
+        "streamId": "streamr.eth/metrics/nodes/firehose/min"
+    },
+    {
+        "duration": 3600000,
+        "streamId": "streamr.eth/metrics/nodes/firehose/hour"
+    },
+    {
+        "duration": 86400000,
+        "streamId": "streamr.eth/metrics/nodes/firehose/day"
+    }
+]
+
+const getPeriodConfig = (rootConfig: StrictStreamrClientConfig): MetricsPeriodConfig[] => {
+    switch (rootConfig.metrics) {
+        case true:
+            return DEFAULT_PERIODS
+        case false:
+            return []
+        default:
+            return rootConfig.metrics.periods
+    }
+}
+
+@scoped(Lifecycle.ContainerScoped)
+export class MetricsPublisher {
+
+    private publisher: Publisher
+    private brubeckNode: BrubeckNode
+    private eventEmitter: StreamrClientEventEmitter
+    private destroySignal: DestroySignal
+    private periodConfigs: MetricsPeriodConfig[]
+    private producers: { stop: () => void}[] = []
+
+    constructor(
+        @inject(Publisher) publisher: Publisher,
+        @inject(BrubeckNode) brubeckNode: BrubeckNode,
+        @inject(StreamrClientEventEmitter) eventEmitter: StreamrClientEventEmitter,
+        @inject(DestroySignal) destroySignal: DestroySignal,
+        @inject(ConfigInjectionToken.Root) rootConfig: StrictStreamrClientConfig
+    ) {
+        this.publisher = publisher
+        this.brubeckNode = brubeckNode
+        this.eventEmitter = eventEmitter
+        this.destroySignal = destroySignal
+        this.periodConfigs = getPeriodConfig(rootConfig)
+    }
+
+    init() {
```

Comment:
Was there a reason not to do this stuff in constructor?
```suggestion
    init(): void {
```